sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -673,6 +673,110 @@ case class Encode(value: Expression, charset: Expression)
 }
 
 /**
+ * Replace all substrings of str that match regexp with rep
+ */
+case class RegExpReplace(subject: Expression, regexp: Expression, rep: Expression)
```

Comment:
It's probably ok for now. I don't think we have that many, and unlike Binary, there are no good names for the arguments. (I think you meant ternary?)